File: zad3/PamoLab/app/src/main/java/com/example/pamolab/ui/BMI/BmiViewModel.java
package com.example.pamolab.ui.BMI;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
public class BmiViewModel extends ViewModel {


    private final MutableLiveData<Integer> mWeight;
    private final MutableLiveData<Integer> mHeight;

    public BmiViewModel() {
        mWeight = new MutableLiveData<>();
        mHeight = new MutableLiveData<>();

    }


    public LiveData<Integer> getWeight() {
        return mWeight;
    }

    public LiveData<Integer> getHeight() {
        return mHeight;
    }

    public void setWeight(int weight) {
        mWeight.setValue(weight);
    }

    public void setHeight(int height) {
        mHeight.setValue(height);
    }

}

File: zad4/PamoLab/app/src/main/java/com/example/pamolab/ui/BMI/BmiViewModel.kt
package com.example.pamolab.ui.BMI

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class BmiViewModel : ViewModel() {
    private val mWeight: MutableLiveData<Int>
    private val mHeight: MutableLiveData<Int>

    init {
        mWeight = MutableLiveData()
        mHeight = MutableLiveData()
    }

    val weight: LiveData<Int>
        get() = mWeight
    val height: LiveData<Int>
        get() = mHeight

    fun setWeight(weight: Int) {
        mWeight.value = weight
    }

    fun setHeight(height: Int) {
        mHeight.value = height
    }
}